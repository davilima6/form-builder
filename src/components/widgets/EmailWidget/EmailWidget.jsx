import * as React from 'react';
import styled from 'styled-components';
import { normalize } from '../../../utils';

const WidgetWrapper = styled.div`
  background-color: #fff;
`;

const Label = styled.label`
  .label {
    font-variant: small-caps;
  }
`;

const Error = styled.div`
  border: 1px dashed red;
`;

type Props = {
  id: string,
  label: string,
  data?: string,
  errors?: Array<Object>,
  onChange: Function,
};

function EmailWidget({ id, label, data, errors, onChange }: Props) {
  const fieldId = `field-${id}`;

  return (
    <WidgetWrapper>
      <Label htmlFor={fieldId}>
        <span className="label">{label}</span>
        <input
          type="email"
          id={fieldId}
          name={id}
          value={data || ''}
          onChange={({ target }) =>
            onChange(id, target.value !== '' ? target.value : undefined)
          }
        />
      </Label>
      {errors.map(message => (
        <Error key={normalize(message)}>{message}</Error>
      ))}
    </WidgetWrapper>
  );
}

EmailWidget.defaultProps = {
  data: '',
  errors: [],
};

export default EmailWidget;
