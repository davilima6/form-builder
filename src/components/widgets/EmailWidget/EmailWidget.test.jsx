import React from 'react';
import { cleanup, render } from 'react-testing-library';
import 'jest-dom/extend-expect';
// import 'jest-styled-components';
import EmailWidget from './EmailWidget';

describe('EmailWidget', () => {
  let component;

  beforeEach(() => {
    const props = {
      label: 'Date of Birth',
      type: 'text',
      data: 'john@google.com',
      errors: [],
      onChange: () => {},
    };

    cleanup();
    component = render(<EmailWidget {...props} />);
  });

  test('renders without crashing', () => expect(component).toBeDefined());

  test('matches snapshot', () => {
    const { asFragment } = component;

    expect(asFragment()).toMatchSnapshot();
    // expect(asFragment()).toHaveStyleRule('background', '#fff');
  });
});
