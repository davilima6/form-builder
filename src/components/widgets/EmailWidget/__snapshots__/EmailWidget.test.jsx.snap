// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailWidget matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #fff;
}

.c1 .label {
  font-variant: small-caps;
}

<div
    class="c0"
  >
    <label
      class="c1"
      for="field-undefined"
    >
      <span
        class="label"
      >
        Date of Birth
      </span>
      <input
        id="field-undefined"
        type="email"
        value="Thu Jan 01 1970 01:00:00 GMT+0100 (Central European Standard Time)"
      />
    </label>
  </div>
</DocumentFragment>
`;
